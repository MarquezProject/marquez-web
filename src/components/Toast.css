@value deepRed, vibrantGreen from "../global_styles.css";

.container {
  position: fixed;
  bottom: 0;
  left: 30%;
  border-radius: 3px;
  color: white;
  padding: 1rem;
  max-width: 40%;
  min-width: 40%;
  text-align: center;
}

.container p {
  white-space: pre-wrap;
}

.error {
  background-color: deepRed;
}

.success {
  background-color: vibrantGreen;
}