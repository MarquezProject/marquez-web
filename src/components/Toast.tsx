import * as React from 'react'
const styles = require('./Toast.css')

interface IProps {
  error: string
  success: string
}

const Toast = (props: IProps) =>
  props.error || props.success ? (
    <div
      className={`${styles.container} ${
        props.error ? styles.error : styles.success
      } shadow animated faster bounceInUp`}
    >
      <p>{props.error || props.success}</p>
    </div>
  ) : null

export default Toast
