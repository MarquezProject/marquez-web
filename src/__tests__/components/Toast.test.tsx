import * as React from 'react'
import { shallow } from 'enzyme'
import Toast from '../../components/Toast'

describe('MainContainer Component', () => {
  it('does not render when there is no error', () => {
    const wrapper = shallow(<Toast error='' />)
    expect(wrapper.find('div')).toHaveLength(0)
  })

  it('renders when there is an error', () => {
    const wrapper = shallow(<Toast error='Some error text' />)
    expect(wrapper.find('div')).toHaveLength(1)
  })  

  it('renders snapshots, too', () => {
    const wrapper = shallow(<Toast error='' />)
    expect(wrapper).toMatchSnapshot()
  })
})
