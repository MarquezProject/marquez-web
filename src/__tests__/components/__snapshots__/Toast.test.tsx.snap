// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer Component renders snapshots, too 1`] = `""`;
